src/sagemaker_containers/_mpi.py

```diff
@@ -0,0 +1,236 @@
+            host_list = self._hosts
+        else:
+            host_list = ['%s:%s' % (host, self._process_per_host) for host in self._hosts]
+
+        msg = 'Env Hosts: %s Hosts: %s process_per_hosts: %s num_processes: %s'
+        logger.info(msg, self._hosts, host_list, self._process_per_host, num_processes)
+
+        overridden_known_options, additional_options = _parse_custom_mpi_options(
+            self._custom_mpi_options)
+
+        logger.info("Network interface name: %s" % self._network_interface_name)
+
+        # TODO(mvs): explain MPI setttings
+        command = ['mpirun',
+                   '--host', ','.join(host_list),
+                   '-np', str(num_processes),
+
+                   '--allow-run-as-root',
+                   '--display-map',
+                   '--tag-output',
+
+                   '-mca', 'btl_tcp_if_include', self._network_interface_name,
+                   '-mca', 'oob_tcp_if_include', self._network_interface_name,
+                   '-mca', 'plm_rsh_no_tree_spawn', '1',
+                   '-bind-to', 'socket', '-map-by', 'slot',
+                   '-mca', 'pml', 'ob1', '-mca', 'btl', '^openib',
+                   '-mca', 'orte_abort_on_non_zero_status', '1',
+
+                   '-x', 'NCCL_MIN_NRINGS=4',
+                   '-x', 'NCCL_SOCKET_IFNAME=%s' % self._network_interface_name,
+                   '-x', 'NCCL_DEBUG=%s' % overridden_known_options.NCCL_DEBUG,
+                   '-x', 'LD_LIBRARY_PATH',
+                   '-x', 'PATH',
+                   '-x', 'LD_PRELOAD=%s' % inspect.getfile(libchangehostname),
+
+                   ]
+
+        command.extend(additional_options)
+
+        for credential in ['AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'AWS_SESSION_TOKEN']:
+            if credential in os.environ:
+                command.extend(['-x', credential])
+
+        for name in self._env_vars:
+            command.extend(['-x', name])
+
+        command.extend(super(MasterRunner, self)._create_command())
+
+        return command
+
+
+_SSH_DAEMON_NOT_FOUND_ERROR_MESSAGE = """
+SSH daemon not found, please install SSH to allow MPI to communicate different nodes in cluster.
+
+You can install ssh by running following commands:
+-------------------------------------------------
+
+1. Install SSH via apt-get:
+
+apt-get update && apt-get install -y --no-install-recommends openssh-server && mkdir -p /var/run/sshd
+
+2. SSH login fix. Otherwise user is kicked off after login:
+sed 's@session\\s*required\\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
+
+3. Create SSH key to allow password less ssh between different docker instances:
+mkdir -p /root/.ssh/ && ssh-keygen -q -t rsa -N '' -f /root/.ssh/id_rsa && \
+  cp /root/.ssh/id_rsa.pub /root/.ssh/authorized_keys && \
+  printf "Host *\n  StrictHostKeyChecking no\n" >> /root/.ssh/config
+"""
+
+
+def _start_sshd_daemon():  # type: () -> None
+    sshd_executable = '/usr/sbin/sshd'
```

Comment:
What if sshd is at /usr/local/sbin/sshd instead? Then this will erroneously fail.

I think it would be preferable to try to run sshd and raise a RuntimeError with this message if sshd isn't anywhere on the PATH.